awx/main/migrations/0165_alter_jobevent_host.py

```diff
@@ -0,0 +1,26 @@
+# Generated by Django 3.2.13 on 2022-06-30 19:08
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('main', '0164_remove_inventorysource_update_on_project_update'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='adhoccommandevent',
+            name='host',
+            field=models.ForeignKey(
+                default=None, editable=False, null=True, on_delete=django.db.models.deletion.DO_NOTHING, related_name='ad_hoc_command_events', to='main.host'
+            ),
+        ),
+        migrations.AlterField(
+            model_name='jobevent',
+            name='host',
+            field=models.IntegerField(default=None, editable=False, null=True),
```

Comment:
This has alarm bells going off for me. Does this operation happen at the table level or does it need to touch each row? Modifying the job event table can potentially affect millions of rows so we need to be extremely careful.

Would this help solve the problem without converting the field type? https://docs.djangoproject.com/en/4.0/ref/models/fields/#django.db.models.ForeignKey.db_constraint

Reply:
In the migration sql it looks like this should only be renaming a column:
```
BEGIN;
--
-- Alter field host on adhoccommandevent
--
--
-- Alter field host on jobevent
--
DROP INDEX IF EXISTS "main_jobevent_host_id_idx1";
ALTER TABLE "main_jobevent" RENAME COLUMN "host_id" TO "host";
COMMIT;
```

That being said we could try leaving this as a ForeignKey and just applying the no-cascade option along with the db_contraint and that should give an identical effect (and likely less code changes).